File: pos_product_stock/models/pos_session.py
from odoo import api, models


class PosSession(models.Model):
    """inherit pos.session to add fields and modules in session."""

    _inherit = "pos.session"

    @api.model
    def _load_pos_data_models(self, config_id):
        """The list of models to be loaded for POS data."""
        data = super()._load_pos_data_models(config_id)
        data += ["stock.location", "stock.quant", "stock.move.line"]
        return data

File: pos_product_stock/static/src/js/pos_session.js
/** @odoo-module */
import {patch} from "@web/core/utils/patch";
import {PosStore} from "@point_of_sale/app/services/pos_store";

patch(PosStore.prototype, {
    async processServerData(data) {
        super.processServerData(...arguments);
        this.stock_quant = this.models["stock.quant"].getAll();
        this.move_line = this.models["stock.move.line"].getAll();
        this.product_product = this.models["product.product"].getAll();
    },
});
